### PR TITLE
MINOR KAFKA-7705 : update java doc for delivery.timeout.ms

### PR DESCRIPTION
**Pull request by hackerwin7**
update KafkaProducer javadoc to put delivery.timeout.ms >= request.timeout.ms + linger.ms

### CODE REVIEW

**Comment by huxihx** [comment]
LGTM. @hachikuji Please help review this minor doc change.

File: clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java [hunk 1]
```diff
@@ -96,7 +96,7 @@
  * Properties props = new Properties();
  * props.put("bootstrap.servers", "localhost:9092");
  * props.put("acks", "all");
- * props.put("delivery.timeout.ms", 30000);
+ * props.put("delivery.timeout.ms", 30001);
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java]: I think we should actually make this example simpler. Most of these configs don't need to be changed. We could change the example to be something like:

```java
 * Properties props = new Properties();
 * props.put("bootstrap.servers", "localhost:9092");
 * props.put("acks", "all");
 *
 * try (Producer<String, String> producer = new KafkaProducer<>(props, new StringSerializer(), new StringSerializer())) {
 *     for (int i = 0; i < 100; i++)
 *         producer.send(new ProducerRecord<>("my-topic", Integer.toString(i), Integer.toString(i)));
 * }
```

What do you think?

Comment (hackerwin7) [reply clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java]: Okay, I'll update it.

**Comment by hackerwin7** [comment]
@ijuma could you review this again? thanks.

**Review by hachikuji — APPROVED** [review]
LGTM. Thanks for the PR!

**Pull request merged**